docs/admin/ai-gateway_integration.md

```diff
@@ -0,0 +1,405 @@
+- **Authentication and Authorization**: Implements API key validation to secure communication.
+- **Backend Security Policy**: Introduces fine-grained access control for backend services, controlling LLM/AI backend usage using token-per-second (TPS) policies to prevent overuse.
+- **Multi-Upstream Provider Support for LLM/AI Services**: The ability to receive requests in the format of one LLM provider and route them to different upstream providers, ensuring compatibility with their expected formats. This is achieved through built-in transformation capabilities that adapt requests and responses accordingly.
+- **AWS Request Signing**: Supports external processing for secure communication with AWS-hosted LLM/AI services.
+
+For more information, see the [Envoy AI Gateway documentation](https://aigateway.envoyproxy.io/docs/).
+
+## Prerequisites
+
+- A Kubernetes cluster with [KServe with Gateway API Enabled](kubernetes_deployment.md), [Envoy Gateway](https://aigateway.envoyproxy.io/docs/getting-started/prerequisites), and [Envoy AI Gateway installed](https://aigateway.envoyproxy.io/docs/getting-started/installation).
+- The `kubectl` command-line tool installed
+
+## Create Namespace
+
+Create a namespace for the InferenceService.
+
+```shell
+kubectl create namespace kserve-test
+```
+
+## Create an InferenceService
+
+Create a secret with the Hugging Face token.
+
+```yaml title="hf-secret.yaml"
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hf-secret
+  namespace: kserve-test
+type: Opaque
+stringData:
+  HF_TOKEN: <token>
+```
+
+Create an InferenceService in the `kserve-test` namespace with OpenAI route prefix disabled. The following example creates an InferenceService with the 'llama3.2-1B' model from Hugging Face.
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: llama3-1b
+  namespace: kserve-test
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: huggingface
+      args:
+        - --model_name=llama3-1b
+        - --model_id=meta-llama/Llama-3.2-1B-Instruct
+      env:
+        - name: HF_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: hf-secret
+              key: HF_TOKEN
+              optional: false
+
+        - name: KSERVE_OPENAI_ROUTE_PREFIX # Disable OpenAI Route Prefix
+          value: ""
+      resources:
+        limits:
+          cpu: "4"
+          memory: 12Gi
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "6"
+          memory: 12Gi
+          nvidia.com/gpu: "1"
+EOF
+```
+
+## Create BackendSecurityPolicy
+
+You can configure the BackendSecurityPolicy for authentication and authorization with the InferenceService. For example, you can create a BackendSecurityPolicy to secure communication with the InferenceService using an API key. But for simplicity, we will ignore the authentication and authorization for this example.
+
+```yaml
+apiVersion: aigateway.envoyproxy.io/v1alpha1
```

Comment:
Some manifests use `kubectl apply -f - <<EOF`, some don't. Wouldn't it be better to use one way?